src/Compilers/CSharp/Portable/Binder/LocalScopeBinder.cs

```diff
@@ -226,6 +226,17 @@ protected ImmutableArray<LocalSymbol> BuildLocals(SyntaxList<StatementSyntax> st
                                 ExpressionVariableFinder.FindExpressionVariables(this, locals, vdecl, localDeclarationBinder);
                             }
 
+                            if (decl.Declaration.Type is ArrayTypeSyntax arrayTypeSyntax)
+                            {
+                                foreach (var rankSpecifier in arrayTypeSyntax.RankSpecifiers)
+                                {
+                                    foreach (var expression in rankSpecifier.Sizes)
+                                    {
+                                        ExpressionVariableFinder.FindExpressionVariables(this, locals, expression, enclosingBinder.GetBinder(expression) ?? enclosingBinder);
```

Comment:
>enclosingBinder.GetBinder(expression) ?? enclosingBinder [](start = 115, length = 56)

Why the binder is different from the one used for the bracketed argument lists and the initializers above?

Reply:
I don't know enough about this part of the codebase.i assumed that  each 
syntax  type required a different binder based on my initial reading of the code, but if I am incorrect I can fix that.

Reply:
>I don't know enough about this part of the codebase.i assumed that  each 
> syntax  type required a different binder based on my initial reading of the code, but if I am incorrect I can fix that.

This binder should match the binder that will be used to bind the expression. Binders are created by the LocalBinderFactory, I do not see us creating any special binders for LocalDeclarationStatementSyntax. BTW, it looks like we need to adjust LocalBinderFactory.VisitLocalDeclarationStatement to visit expressions that we are planning to bind now.

---
In reply to: [255651969](https://github.com/dotnet/roslyn/pull/32843#discussion_r255651969) [](ancestors = 255651969)